controller/api/destination/watcher/ip_watcher.go

```diff
@@ -93,6 +95,40 @@ func NewIPWatcher(k8sAPI *k8s.API, endpoints *EndpointsWatcher, log *logging.Ent
 		},
 	})
 
+	k8sAPI.Pod().Informer().AddIndexers(cache.Indexers{hostIPIndex: func(obj interface{}) ([]string, error) {
+		if pod, ok := obj.(*corev1.Pod); ok {
+			var hostIPPods []string
+			if pod.Status.HostIP != "" {
+				// If the pod runs in the host network, then all of it's
+				// containers' ports should be added to the indexer.
+				//
+				// If the pod does not run in the host network but does have a
+				// host IP, then one or more of it's containers should specify
+				// a host port. Each hostIP:hostPort address should be added
+				// to the indexer.
+				if pod.Spec.HostNetwork {
```

Comment:
I thought this was necessary, but upon further reading of the docs, I think we can actually omit it.  

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#containerport-v1-core

Specifically, it says that if HostNetwork is specified then containerPort MUST match hostPort.  Therefore, if HostNetwork is specified then hostPort will be non-zero and will be properly handled by the case below.